packages/s3-store/index.ts

```diff
@@ -82,6 +125,7 @@ export class S3Store extends DataStore {
   private preferredPartSize: number
   private expirationPeriodInMilliseconds = 0
   private useTags = true
+  private semaphore: Semaphore
```

Comment:
```suggestion
  private partUploadSemaphore: Semaphore
```

A good name helps understand what the semaphore actually controls.